x-pack/plugins/ml/server/lib/alerts/register_ml_alerts.ts

```diff
@@ -19,7 +20,9 @@ export interface RegisterAlertParams {
   mlServicesProviders: MlServicesProviders;
 }
 
-export function registerMlAlerts(alertParams: RegisterAlertParams) {
-  registerAnomalyDetectionAlertType(alertParams);
-  registerJobsMonitoringRuleType(alertParams);
+export function registerMlAlerts(alertParams: RegisterAlertParams, enabledFeatures: MlFeatures) {
+  if (enabledFeatures.ad === true) {
```

Comment:
question/suggestion: did you consider performing a check before `registerMlAlerts` and `registerCasesPersistableState` are called?

Reply:
I'm passing in the enabled features into `registerMlAlerts` and have the check happen in there.
I did it this way to allow for future non AD alerts to still be registered by this function. As in, the `registerMlAlerts` function will decide which alerts to register.

Reply:
To move the check out into the plugin.ts file, these register functions would need to be renamed to have `AD` or similar in the name e.g. `registerMlADAlerts`
I thought it would be neater to have one function to register all ML alerts, and have it decide which ones to register based on feature enablement.